elytron/src/main/java/org/wildfly/extension/elytron/AggregateRealmDefinition.java

```diff
@@ -122,18 +131,26 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
             final List<InjectedValue<SecurityRealm>> authorizationRealmValues = new ArrayList<>();
             ModelNode authorizationRealmNode = AUTHORIZATION_REALM.resolveModelAttribute(context, model);
 
+            String principalTransformer = PRINCIPAL_TRANSFORMER.resolveModelAttribute(context, model).asStringOrNull();
+            final InjectedValue<PrincipalTransformer> principalTransformerValue = new InjectedValue<PrincipalTransformer>();
+
+            String principalTransformerRuntimeCapability = RuntimeCapability.buildDynamicCapabilityName(PRINCIPAL_TRANSFORMER_CAPABILITY, principalTransformer);
+            ServiceName principalTransformerServiceName = context.getCapabilityServiceName(principalTransformerRuntimeCapability, PrincipalTransformer.class);
+
             TrivialService<SecurityRealm> aggregateRealmService = new TrivialService<SecurityRealm>(() -> {
                 SecurityRealm[] authorizationRealms = new SecurityRealm[authorizationRealmValues.size()];
                 for (int i = 0; i < authorizationRealms.length; i++) {
                     authorizationRealms[i] = authorizationRealmValues.get(i).getValue();
                 }
 
-                return new AggregateSecurityRealm(authenticationRealmValue.getValue(), authorizationRealms);
+                return new AggregateSecurityRealm(authenticationRealmValue.getValue(), principalTransformerValue.getValue(), authorizationRealms);
```

Comment:
There should be a null check here as well so that ```principalTransformerValue.getValue()``` is only called if ```principalTransformerValue``` is not null. Otherwise, the existing constructor can be used.

Reply:
Fixed